rcl/fibonacci_action_client/main.c

```diff
@@ -0,0 +1,165 @@
+  rcl_wait_set_t wait_set = rcl_get_zero_initialized_wait_set();
+  rv = rcl_wait_set_init(&wait_set, num_subscriptions, num_guard_conditions, num_timers, num_clients, num_services, 0, &context, rcl_get_default_allocator());
+  if (RCL_RET_OK != rv) {
+    printf("Wait set initialization error: %s\n", rcl_get_error_string().str);
+    return 1;
+  }
+
+  int64_t goal_sequence_number;
+  int64_t result_sequence_number;
+
+  bool done = false;
+  bool goal_accepted = false;
+  int order = 10;
+
+  sleep(2);
+  printf("Sending goal request\n");
+  example_interfaces__action__Fibonacci_SendGoal_Request ros_goal_request;
+  ros_goal_request.goal.order = order;
+  rcl_action_send_goal_request(&action_client, &ros_goal_request, &goal_sequence_number);
+  printf("Request sent\n");
+
+  do {    
+    rv = rcl_wait_set_clear(&wait_set);
+    if (RCL_RET_OK != rv) {
+      printf("Wait set clear error: %s\n", rcl_get_error_string().str);
+      return 1;
+    }
+    
+    size_t client_index, subscription_index;
+    rcl_action_wait_set_add_action_client(&wait_set, &action_client, &client_index, &subscription_index);
+    
+    rcl_wait(&wait_set, RCL_MS_TO_NS(50));
+
+    bool is_feedback_ready = false;
+    bool is_status_ready = false;
+    bool is_goal_response_ready = false;
+    bool is_cancel_response_ready = false;
+    bool is_result_response_ready = false;
+
+    rcl_action_client_wait_set_get_entities_ready(
+          &wait_set,
+          &action_client,
+          &is_feedback_ready,
+          &is_status_ready,
+          &is_goal_response_ready,
+          &is_cancel_response_ready,
+          &is_result_response_ready);
+
+    if (is_goal_response_ready)
+    {
+      printf("Goal response ready\n");
+      example_interfaces__action__Fibonacci_SendGoal_Response ros_goal_response;
+      rmw_request_id_t response_header;
+
+      rcl_action_take_goal_response(&action_client, &response_header, &ros_goal_response);
+      if (ros_goal_response.accepted)
+      {
+          printf("Goal request accepted\n");
+          printf("Requesting response...\n");
+
+          goal_accepted = true;
+
+          example_interfaces__action__Fibonacci_GetResult_Request ros_result_request;
+          rcl_action_send_result_request(&action_client, &ros_result_request, &result_sequence_number);
+      }
+    }else if(is_feedback_ready){
+      example_interfaces__action__Fibonacci_FeedbackMessage ros_feedback;
+
+      ros_feedback.feedback.sequence.data = (int32_t*) malloc(order * sizeof(int32_t));
+      ros_feedback.feedback.sequence.capacity = order;
+
+      rcl_action_take_feedback(&action_client, &ros_feedback);
+      
+      printf("Feedback: [");
+      for (size_t i = 0; i < ros_feedback.feedback.sequence.size ; i++)
+      {
+        printf("%d, ", ros_feedback.feedback.sequence.data[i]);
+      }
+      printf("]\n");
+
+      free(ros_feedback.feedback.sequence.data);
+
+    }else if(is_result_response_ready){
```

Comment:
Could it be `is_goal_response_ready`, `is_feedback_ready` and `is_result_response_ready` true at the same time?

Reply:
I think so, I'm refactoring to independant if blocks.